spark-indexer/src/main/java/org/uniprot/store/spark/indexer/proteome/converter/DatasetProteomeEntryConverter.java

```diff
@@ -19,26 +30,338 @@
+                    new ProteomeIdBuilder((row.getString(row.fieldIndex(PANPROTEOME)))).build());
+        }
+        // modified
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        builder.modified(
+                LocalDate.parse(row.getDate(row.fieldIndex(MODIFIED)).toString(), formatter));
+        // proteome type
+        ProteomeType proteomeType = ProteomeType.NORMAL;
+        boolean isReference = row.getBoolean(row.fieldIndex(IS_REFERENCE_PROTEOME));
+        boolean isRepresentative = row.getBoolean(row.fieldIndex(IS_REPRESENTATIVE_PROTEOME));
+
+        if (isReference && isRepresentative) {
+            proteomeType = ProteomeType.REFERENCE_AND_REPRESENTATIVE;
+        } else if (isReference) {
+            proteomeType = ProteomeType.REFERENCE;
+        } else if (isRepresentative) {
+            proteomeType = ProteomeType.REPRESENTATIVE;
+        }
+        builder.proteomeType(proteomeType);
+        // genome annotation
+        if (hasFieldName(GENOME_ANNOTATION, row)) {
+            GenomeAnnotation genomeAnnotation =
+                    getGenomeAnnotation((Row) row.get(row.fieldIndex(GENOME_ANNOTATION)));
+            builder.genomeAnnotation(genomeAnnotation);
+        }
+        // genome assembly
+        if (hasFieldName(GENOME_ASSEMBLY, row)) {
+            GenomeAssembly genomeAssembly =
+                    getGenomeAssembly((Row) row.get(row.fieldIndex(GENOME_ASSEMBLY)));
+            builder.genomeAssembly(genomeAssembly);
+        }
+        // annotation score
+        if (hasFieldName(ANNOTATION_SCORE, row)) {
+            Integer annotationScore =
+                    getAnnotationScore((Row) row.get(row.fieldIndex(ANNOTATION_SCORE)));
+            builder.annotationScore(annotationScore);
+        }
+        // component set
+        List<Row> componentRows = row.getList(row.fieldIndex(COMPONENT));
+        builder.componentsSet(
+                componentRows.stream().map(this::getComponent).collect(Collectors.toList()));
+        // citation set
+        if (hasFieldName(REFERENCE, row)) {
+            List<Row> referenceRows = row.getList(row.fieldIndex(REFERENCE));
+            builder.citationsSet(
+                    referenceRows.stream().map(this::getCitation).collect(Collectors.toList()));
+        }
+        // redundant proteome
+        if (hasFieldName(REDUNDANT_PROTEOME, row)) {
+            List<Row> redundantProteomeRows = row.getList(row.fieldIndex(REDUNDANT_PROTEOME));
+            builder.redundantProteomesSet(
+                    redundantProteomeRows.stream()
+                            .map(this::getRedundantProteome)
+                            .collect(Collectors.toList()));
+        }
+        // excluded
+        if (hasFieldName(EXCLUDED, row)) {
+            ExclusionReason exclusionReason =
+                    getExclusionReason((Row) row.get(row.fieldIndex(EXCLUDED)));
+            builder.exclusionReasonsAdd(exclusionReason);
```

Comment:
We have to change the type here as well. Keep in mind that we also have NORMAL proteome type.